####
Workaround provided by Philip Kovacs, based on an idea found in
the event_forward_to_rootwin() function from xfdesktop:

http://svn.xfce.org/svn/xfce/xfdesktop/trunk/src/main.c

It allows a more consistent menu behavior under Xfce4; see 
http://adesklets.sourceforge.net/forum/viewtopic.php?p=1658#1658
for details.
####
--- src/adesklets.c.old 2006-03-03 00:00:00.000000000 -0500
+++ src/adesklets.c     2006-03-03 00:00:01.000000000 -0500
@@ -2909,6 +2909,7 @@ if(imlib_context_get_filter()) {\
 		  event("menufire 0 %s\n",menu_str);
 #ifdef CONTROL_ON_CONTEXT_MENU
 	      } else {
+		XUngrabPointer(adesklets.display, ev.xbutton.time);
 		ev.xbutton.window = adesklets.root ;
 		XSendEvent(adesklets.display, ev.xbutton.window, 
 			   False, ButtonPressMask, &ev);
